File: frontend/src/api/wellbeing.ts
import { api } from './client';

export interface PulseResponse {
  id: string;
  survey_id: string;
  user_id: string;
  morale: number;
  workload: number;
  support: number;
  comments: string | null;
  created_at: string;
}

export interface PulseSurvey {
  id: string;
  workspace_id: string;
  title: string;
  status: string;
  end_date: string | null;
  responses: PulseResponse[];
  created_at: string;
}

export interface Kudos {
  id: string;
  workspace_id: string;
  from_user_id: string;
  to_user_id: string;
  message: string;
  created_at: string;
}

export const wellbeingApi = {
  // Surveys
  listSurveys: (workspaceId: string) =>
    api.get<PulseSurvey[]>(`/workspaces/${workspaceId}/wellbeing/surveys`),

  createSurvey: (workspaceId: string, data: { title: string; end_date?: string }) =>
    api.post<PulseSurvey>(`/workspaces/${workspaceId}/wellbeing/surveys`, data),

  submitResponse: (workspaceId: string, surveyId: string, data: { morale: number; workload: number; support: number; comments?: string }) =>
    api.post<PulseResponse>(`/workspaces/${workspaceId}/wellbeing/surveys/${surveyId}/respond`, data),

  // Kudos
  listKudos: (workspaceId: string) =>
    api.get<Kudos[]>(`/workspaces/${workspaceId}/wellbeing/kudos`),

  giveKudos: (workspaceId: string, data: { to_user_id: string; message: string }) =>
    api.post<Kudos>(`/workspaces/${workspaceId}/wellbeing/kudos`, data),
};

File: backend/app/models/wellbeing.py
from __future__ import annotations

import uuid
from typing import TYPE_CHECKING

from sqlalchemy import Date, ForeignKey, Integer, String, Text
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.models.base import Base, TimestampMixin, UUIDPrimaryKey

if TYPE_CHECKING:
    from app.models.user import User


class PulseSurvey(Base, UUIDPrimaryKey, TimestampMixin):
    __tablename__ = "pulse_surveys"

    workspace_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("workspaces.id", ondelete="CASCADE"), nullable=False
    )
    title: Mapped[str] = mapped_column(String(255), nullable=False)
    status: Mapped[str] = mapped_column(String(20), default="active", nullable=False)  # active, closed
    end_date: Mapped[str | None] = mapped_column(Date, nullable=True)

    responses: Mapped[list[PulseResponse]] = relationship(back_populates="survey", cascade="all, delete-orphan")


class PulseResponse(Base, UUIDPrimaryKey, TimestampMixin):
    __tablename__ = "pulse_responses"

    survey_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("pulse_surveys.id", ondelete="CASCADE"), nullable=False
    )
    user_id: Mapped[uuid.UUID | None] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id", ondelete="SET NULL"), nullable=True
    )
    morale: Mapped[int | None] = mapped_column(Integer, nullable=True)  # 1-5
    workload: Mapped[int | None] = mapped_column(Integer, nullable=True)  # 1-5
    support: Mapped[int | None] = mapped_column(Integer, nullable=True)  # 1-5
    comments: Mapped[str | None] = mapped_column(Text, nullable=True)

    survey: Mapped[PulseSurvey] = relationship(back_populates="responses")


class Kudos(Base, UUIDPrimaryKey, TimestampMixin):
    __tablename__ = "kudos"

    workspace_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("workspaces.id", ondelete="CASCADE"), nullable=False
    )
    from_user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False
    )
    to_user_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"), nullable=False
    )
    message: Mapped[str] = mapped_column(String(500), nullable=False)

    from_user: Mapped[User] = relationship(foreign_keys=[from_user_id])
    to_user: Mapped[User] = relationship(foreign_keys=[to_user_id])
